source/docs/software/vscode-overview/debugging-robot-program.rst

```diff
@@ -5,72 +5,54 @@ Inevitably, a program will not behave in the way we expect it to behave.  When t
 
 Breakpoints
 -----------
 
-A "breakpoint" is a line of code at which the debugger will stop the program execution so that the user can examine the program state.  This is extremely useful while debugging, as it allows the user to pause the program at specific points in problematic code to determine where exactly the program is deviating from the expected behavior.
+A "breakpoint" is a line of code at which the debugger will pause the program execution so that the user can examine the program state.  This is extremely useful while debugging, as it allows the user to pause the program at specific points in problematic code to determine where exactly the program is deviating from the expected behavior.
+
+The debugger will automatically pause at the first breakpoint it encounters.
 
 Setting a Breakpoint
 ~~~~~~~~~~~~~~~~~~~~
 
 |Setting a Breakpoint|
 
-Double-click in the left margin of the source code window to set a breakpoint in your user program: A small red circle indicates the breakpoint has been set on the corresponding line.
-
-Viewing Program State
-~~~~~~~~~~~~~~~~~~~~~
-
-|Debug Tab|
-
-When the program is stopped at a breakpoint, various pieces of program state can be viewed from the debug tab.  The Debug tab is accessed by clicking on the debug icon on the far left pane.  Within the debug tab, the program state can be viewed from an assortment of "panes."
+Click in the left margin of the source code window (to the left of the line number) to set a breakpoint in your user program: A small red circle indicates the breakpoint has been set on the corresponding line.
 
-The Variables Pane
-^^^^^^^^^^^^^^^^^^
+Debugging with Print Statements
+-------------------------------
 
-|Variables Pane|
+Another way to debug your program is to use print statements in your code and view them using the RioLog in Visual Studio Code or the Driver Station.  Print statements should be added with care as they are not very efficient especially when used in high quantities.  They should be removed for competition as they can cause loop overruns.
 
-The Variables view shows the current values of variables. To see a variable that is not displayed, select the “Watch” pane and enter the variable name. This will show the variable’s value if it’s in-scope. Clicking on the arrows next to a variable name will expand the tree and show its `fields <https://en.wikipedia.org/wiki/Field_(computer_science)>`__.
+.. tabs::
 
-Watch Pane
-^^^^^^^^^^
+    .. code-tab:: java
 
-|Watch Pane|
+        System.out.print("example");
 
-The Watch Pane can be used to monitor specific variables or expressions while debugging. To add an expression, right-click and select Add Expression.
+    .. code-tab:: c++
 
-Call Stack
-^^^^^^^^^^
+        fmt::print("example");
```

Comment:
This is not going to compile unless teams add `fmtlib` as a dependency. We should recommend `wpi::outs() << "example\n";` instead (from the `wpi/raw_ostream` header).

Reply:
blame @calcmogul I got that from him.  Fixed

Reply:
I didn't intend for it to be in the docs. :stuck_out_tongue: